quill-cassandra/src/main/scala/io/getquill/CassandraAsyncContext.scala

```diff
@@ -36,20 +36,23 @@ class CassandraAsyncContext[N <: NamingStrategy](
     super.performIO(io)
   }
 
-  def executeQuery[T](cql: String, prepare: Prepare = identityPrepare, extractor: Extractor[T] = identityExtractor)(implicit ec: ExecutionContext): Future[List[T]] = {
-    val (params, bs) = prepare(super.prepare(cql))
-    logger.logQuery(cql, params)
-    session.executeAsync(bs)
-      .map(_.all.asScala.toList.map(extractor))
-  }
+  def executeQuery[T](cql: String, prepare: Prepare = identityPrepare, extractor: Extractor[T] = identityExtractor)(implicit ec: ExecutionContext): Future[List[T]] =
+    Future(prepare(super.prepare(cql))) flatMap {
```

Comment:
Could you avoid infix syntax? (`.flatMap`)

Reply:
:+1: